src/worker.c

```diff
@@ -379,12 +382,14 @@ worker_main(Datum main_arg)
 							argValues[1] = Int32GetDatum(http_status_code);
 							nulls[1] = ' ';
 
-							argTypes[2] = CSTRINGOID;
-							argValues[2] = CStringGetDatum(cdata->body->data);
-							if(cdata->body->data[0] == '\0')
-								nulls[2] = 'n';
-							else
-								nulls[2] = ' ';
+							argTypes[2] = BYTEAOID;
+
+							body_data = (bytea *) palloc(cdata->body->len + VARHDRSZ);
+							SET_VARSIZE(body_data, cdata->body->len + VARHDRSZ);
+							memcpy(VARDATA(body_data), cdata->body->data, cdata->body->len);
```

Comment:
This way of converting a `char*` to `bytea` is done in many parts of the postgres code([example](https://github.com/postgres/postgres/blob/8709228775b549a2388e6568f463a4115e5a4c10/contrib/pageinspect/heapfuncs.c#L389-L391)).